File: 189-rotate-array/189-rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        k = k % n
        nums[0:n] = nums[0:n][::-1]
        nums[0:k] = nums[0:k][::-1] 
        nums[k:n] = nums[k:n][::-1] 

File: 189-rotate-array/189-rotate-array.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int n = nums.size();
        int tmp;
        k = k % n;
        
        for (int i = 0; i<n/2; i++)
        {
            tmp = nums[i];
            nums[i] = nums[n-i-1];
            nums[n-i-1] = tmp;
        }
        
        for (int i = 0; i<k/2; i++)
        {
            tmp = nums[i];
            nums[i] = nums[k-i-1];
            nums[k-i-1] = tmp;
        }
        
        for (int i = 0; i<(n-k)/2; i++)
        {
            tmp = nums[i+k];
            nums[i+k] = nums[n-i-1];
            nums[n-i-1] = tmp;
        }
        
    }
};